samples/subsys/tracing/ sample.tracing.user fails to build in multiple platforms

**Describe the bug**
samples/subsys/tracing/ sample.tracing.user fails to build in:

* m5stack_atom_lite/esp32/appcpu
* m5stack_atom_lite/esp32/procpu
* xenvm
* esp_wrover_kit/esp32/procpu
* esp_wrover_kit/esp32/appcpu
* intel_adl_rvp
* intel_adl_crb
* yd_esp32/esp32/appcpu
* yd_esp32/esp32/procpu
* xiao_esp32s3/esp32s3/procpu
* xiao_esp32s3/esp32s3/appcpu
* esp32s3_devkitm/esp32s3/appcpu
* esp32s3_devkitm/esp32s3/procpu
* up_squared
* stamp_c3
* esp32_ethernet_kit/esp32/procpu
* icev_wireless
* esp32_ethernet_kit/esp32/appcpu
* esp32_devkitc_wrover/esp32/appcpu
* esp32_devkitc_wrover/esp32/procpu
* odroid_go/esp32/appcpu
* esp32s2_franzininho
* odroid_go/esp32/procpu
* olimex_esp32_evb/esp32/procpu
* olimex_esp32_evb/esp32/appcpu
* heltec_wifi_lora32_v2/esp32/appcpu
* heltec_wifi_lora32_v2/esp32/procpu
* esp32_devkitc_wroom/esp32/procpu
* esp32_devkitc_wroom/esp32/appcpu
* tlsr9518adk80d
* wio_terminal
* litex_vexriscv
* intel_ehl_crb
* intel_rpl_s_crb
* intel_rpl_p_crb
* intel_ehl_crb/elkhart_lake/sbl
* heltec_wireless_stick_lite_v3/esp32s3/appcpu
* heltec_wireless_stick_lite_v3/esp32s3/procpu
* m5stack_atoms3/esp32s3/procpu
* m5stack_atoms3/esp32s3/appcpu
* esp32s3_touch_lcd_1_28/esp32s3/appcpu
* esp32s3_touch_lcd_1_28/esp32s3/procpu
* esp32s2_devkitc
* kincony_kc868_a32/esp32/appcpu
* kincony_kc868_a32/esp32/procpu
* fvp_base_revc_2xaemv8a//smp/ns
* esp32s3_devkitc/esp32s3/appcpu
* esp32s3_devkitc/esp32s3/procpu
* m5stack_atoms3_lite/esp32s3/appcpu
* m5stack_atoms3_lite/esp32s3/procpu
* m5stack_core2/esp32/procpu
* m5stack_core2/esp32/appcpu
* esp32s3_luatos_core/esp32s3/procpu/usb
* esp32s3_luatos_core/esp32s3/procpu
* esp32s3_luatos_core/esp32s3/appcpu
* esp32s3_luatos_core/esp32s3/appcpu/usb
* xiao_esp32c3
* ast1030_evb
* esp32c3_devkitm
* esp32c3_luatos_core/esp32c3/usb
* esp32c3_luatos_core/esp32c3
* m5stickc_plus/esp32/appcpu
* m5stickc_plus/esp32/procpu
* m5stack_stamps3/esp32s3/procpu
* m5stack_stamps3/esp32s3/appcpu
* cy8cproto_062_4343w
* esp32s2_lolin_mini
* up_squared_pro_7000
* titanium_ti60_f225
* esp32s2_saola

**To Reproduce**
Steps to reproduce the behavior:
1. twister -T samples/subsys/tracing/ -s sample.tracing.user --all

OR 

1. mkdir build; cd build
2. cmake -GNinja -DBOARD=up_squared  ../samples/subsys/tracing/ -DCONF_FILE="prj_user.conf"
3. ninja
4. See error

**Expected behavior**
No build errors

**Impact**
70 build failures in weekly CI

**Logs and console output**
```
./kernel/mempool.c: In function 'k_realloc':
./kernel/mempool.c:132:8: error: macro "sys_port_trace_k_heap_sys_k_realloc_enter" passed 2 arguments, but takes just 1
  132 |         SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_heap_sys, k_realloc, heap, ptr);
      | ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ./include/zephyr/tracing/tracing.h:20,
                 from ./include/zephyr/kernel.h:6155,
                 from ./kernel/mempool.c:7:
./subsys/tracing/user/./tracing_user.h:301: note: macro "sys_port_trace_k_heap_sys_k_realloc_enter" defined here
  301 | #define sys_port_trace_k_heap_sys_k_realloc_enter(heap)
      | 
In file included from ./include/zephyr/kernel.h:22:
./include/zephyr/tracing/tracing_macros.h:55:9: error: 'sys_port_trace_k_heap_sys_k_realloc_enter' undeclared (first use in this function)
   55 |         sys_port_trace_ ## name ## _ ## func ## _enter
      |         ^~~~~~~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:155:65: note: in definition of macro 'sys_port_trace_type_mask_k_heap_sys'
  155 |         #define sys_port_trace_type_mask_k_heap_sys(trace_call) trace_call
      |                                                                 ^~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:310:17: note: in expansion of macro 'SYS_PORT_TRACING_TYPE_MASK'
  310 |                 SYS_PORT_TRACING_TYPE_MASK(obj_type, \
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:311:25: note: in expansion of macro '_SYS_PORT_TRACING_OBJ_FUNC_ENTER'
  311 |                         _SYS_PORT_TRACING_OBJ_FUNC_ENTER(obj_type, func)(obj, ##__VA_ARGS__)); \
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./kernel/mempool.c:132:9: note: in expansion of macro 'SYS_PORT_TRACING_OBJ_FUNC_ENTER'
  132 |         SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_heap_sys, k_realloc, heap, ptr);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:55:9: note: each undeclared identifier is reported only once for each function it appears in
   55 |         sys_port_trace_ ## name ## _ ## func ## _enter
      |         ^~~~~~~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:155:65: note: in definition of macro 'sys_port_trace_type_mask_k_heap_sys'
  155 |         #define sys_port_trace_type_mask_k_heap_sys(trace_call) trace_call
      |                                                                 ^~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:310:17: note: in expansion of macro 'SYS_PORT_TRACING_TYPE_MASK'
  310 |                 SYS_PORT_TRACING_TYPE_MASK(obj_type, \
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:311:25: note: in expansion of macro '_SYS_PORT_TRACING_OBJ_FUNC_ENTER'
  311 |                         _SYS_PORT_TRACING_OBJ_FUNC_ENTER(obj_type, func)(obj, ##__VA_ARGS__)); \
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./kernel/mempool.c:132:9: note: in expansion of macro 'SYS_PORT_TRACING_OBJ_FUNC_ENTER'
  132 |         SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_heap_sys, k_realloc, heap, ptr);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./kernel/mempool.c:135:8: error: macro "sys_port_trace_k_heap_sys_k_realloc_exit" passed 3 arguments, but takes just 2
  135 |                 SYS_PORT_TRACING_OBJ_FUNC_EXIT(k_heap_sys, k_realloc, heap, ptr, NULL);
      | ^               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./subsys/tracing/user/./tracing_user.h:302: note: macro "sys_port_trace_k_heap_sys_k_realloc_exit" defined here
  302 | #define sys_port_trace_k_heap_sys_k_realloc_exit(heap, ret)
      | 
./include/zephyr/tracing/tracing_macros.h:59:9: error: 'sys_port_trace_k_heap_sys_k_realloc_exit' undeclared (first use in this function)
   59 |         sys_port_trace_ ## name ## _ ## func ## _exit
      |         ^~~~~~~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:155:65: note: in definition of macro 'sys_port_trace_type_mask_k_heap_sys'
  155 |         #define sys_port_trace_type_mask_k_heap_sys(trace_call) trace_call
      |                                                                 ^~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:345:17: note: in expansion of macro 'SYS_PORT_TRACING_TYPE_MASK'
  345 |                 SYS_PORT_TRACING_TYPE_MASK(obj_type, \
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
./include/zephyr/tracing/tracing_macros.h:346:25: note: in expansion of macro '_SYS_PORT_TRACING_OBJ_FUNC_EXIT'
  346 |                         _SYS_PORT_TRACING_OBJ_FUNC_EXIT(obj_type, func)(obj, ##__VA_ARGS__)); \
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./kernel/mempool.c:135:17: note: in expansion of macro 'SYS_PORT_TRACING_OBJ_FUNC_EXIT'
  135 |                 SYS_PORT_TRACING_OBJ_FUNC_EXIT(k_heap_sys, k_realloc, heap, ptr, NULL);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./kernel/mempool.c:146:8: error: macro "sys_port_trace_k_heap_sys_k_realloc_exit" passed 3 arguments, but takes just 2
  146 |         SYS_PORT_TRACING_OBJ_FUNC_EXIT(k_heap_sys, k_realloc, heap, ptr, ret);
      | ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./subsys/tracing/user/./tracing_user.h:302: note: macro "sys_port_trace_k_heap_sys_k_realloc_exit" defined here
  302 | #define sys_port_trace_k_heap_sys_k_realloc_exit(heap, ret)
```
https://github.com/zephyrproject-rtos/zephyr/runs/25996829280

**Environment (please complete the following information):**
 - OS: Linux Ubuntu 22.04
 - Toolchain (e.g Zephyr SDK, ...): 0.16.8
 - Commit SHA or Version used: Main as of now b46fb464cbaaad8d094f75487a3502a11126d025

**Additional context**
https://github.com/zephyrproject-rtos/zephyr/pull/72483

--- a/subsys/tracing/user/tracing_user.h
+++ b/subsys/tracing/user/tracing_user.h
@@ -298,8 +298,8 @@ void sys_trace_idle(void);
 #define sys_port_trace_k_heap_sys_k_free_exit(heap, heap_ref)
 #define sys_port_trace_k_heap_sys_k_calloc_enter(heap)
 #define sys_port_trace_k_heap_sys_k_calloc_exit(heap, ret)
-#define sys_port_trace_k_heap_sys_k_realloc_enter(heap)
-#define sys_port_trace_k_heap_sys_k_realloc_exit(heap, ret)
+#define sys_port_trace_k_heap_sys_k_realloc_enter(heap, ptr)
+#define sys_port_trace_k_heap_sys_k_realloc_exit(heap, ptr, ret)
 
 #define sys_port_trace_k_mem_slab_init(slab, rc)
 #define sys_port_trace_k_mem_slab_alloc_enter(slab, timeout)

